akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocketClientBlueprint.scala

```diff
@@ -53,7 +53,7 @@ object WebSocketClientBlueprint {
     val valve = StreamUtils.OneTimeValve()
 
     val (initialRequest, key) = Handshake.Client.buildRequest(uri, extraHeaders, subprotocol.toList, settings.websocketRandomFactory())
-    val hostHeader = Host(uri.authority)
+    val hostHeader = Host(uri.authority.normalizedFor(uri.scheme))
```

Comment:
Not sure where that would be? For outgoing connections in `Http.sharedGateway` it already uses `effectivePort`, for incoming you don't use Uri to bind it.

Maybe this should have been `effectivePort` though.